app/busy.go

```diff
@@ -7,14 +7,25 @@ import (
 	"sync"
 	"sync/atomic"
 	"time"
+
+	"github.com/mattermost/mattermost-server/v5/einterfaces"
+	"github.com/mattermost/mattermost-server/v5/model"
 )
 
-type Busy struct {
-	busy int32 // protected via atomic for fast IsBusy calls
+const (
+	TIMESTAMP_FORMAT = "Mon Jan 2 15:04:05 -0700 MST 2006"
+)
 
+// Busy represents the busy state of the server. A server marked busy
+// will have non-critical services disabled. If a Cluster is provided
+// any changes will be propagated to each node.
+type Busy struct {
+	busy    int32 // protected via atomic for fast IsBusy calls
 	mux     sync.RWMutex
 	timer   *time.Timer
 	expires time.Time
+
+	Cluster einterfaces.ClusterInterface
```

Comment:
The unique reason to have it public is the absence of a way to initialize it. I would create a constructor function (like `NewBusy`, and make the `cluster` attribute private.

Reply:
Busy.Cluster now un-exported via https://github.com/mattermost/mattermost-server/pull/13267/commits/2f6c11caca415f6b786c2e3ead3abbbeaa981780